crates/cli/src/main.rs

```diff
@@ -651,22 +669,31 @@ impl Commands {
                                         }
                                         Some(OnFail::Stop) | None if !exit_status.success() => {
                                             error!("Build failed for {long_name} with status {exit_status}");
-                                            std::process::exit(1);
+                                            exit_code = 1;
+                                            break 'run_groups;
                                         }
                                         _ => {
                                             info!("{long_name} finished with status {exit_status}");
                                         }
                                     },
                                     Err(err) => {
                                         error!("{err}");
-                                        std::process::exit(1);
+                                        exit_code = 1;
+                                        break 'run_groups;
                                     }
                                 }
                             }
                         }
                         Err(e) => bail!(e),
                     }
                 }
+
+                #[cfg(feature = "telemetry")]
+                if let Some(join) = telem_join {
+                    join.await.ok();
+                }
+
+                std::process::exit(exit_code)
```

Comment:
Can you do this only if `!= 1` slightly pedantic :)

Reply:
thought about it, sure